packages/binding-http/test/http-client-test.ts

```diff
@@ -463,7 +464,9 @@ class HttpClientTest2 {
         };
 
         client
-            .subscribeResource(form, (data) => {
+            .subscribeResource(form, async (data) => {
+                const str = await text(data.body); // SHOULD be "Test event"
+                str.should.eql("Test event");
```

Comment:
Question: IF I add a _wrong_ test like

`str.should.eql("Test event XXX");`

I get the following error

`Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called;`

How can I fix that?

Reply:
Mhh, what you proposed didn't work for me. I played around but didn't manage to succeed.

Anyhow, I found another clumsy code that manages to detect any future error, see https://github.com/eclipse-thingweb/node-wot/pull/1298/commits/2808a7f2ab67315ef55028007bf16f752dcb75c3

Reply:
I'll give it a try to fix it tomorrow if I don't suceed let's go with your solution.

Reply:
I know, the code is not very "nice" but it does the job 🤷‍♂️🙈

Reply:
Ok I dag deeper into the issue, I suggest to remove the code added in 2808a7f and update npm run test adding this env variable: NODE_OPTIONS=--unhandled-rejections=strict. As explained in https://github.com/mochajs/mocha/issues/2640#issuecomment-1663388547. It seems that mocha does ignore the new nodejs mechanism of reporting async unhandled exceptions.

Reply:
I can confirm that setting the env varaible works. I also _quickly_ checked how this can be done in a cross environment manner (`set NODE_OPTIONS=bla` on Windows vs. `export NODE_OPTIONS=bla` on Linux/Mac). There is no really good solution I think.

There is also [cross-env package](https://www.npmjs.com/package/cross-env) which is in maintenance mode.

Anyhow, all this makes me think this is becoming overly complex and not worth it for this problem.

Any thoughts and/or ideas?

Reply:
mocha relays node options to the node runner so you add the option to the command easily:
`mocha --unhandled-rejections=strict --require ts-node/register --extension ts`

Reply:
a bit weird cause I did not spot those errors when testing it (only if the SSE test was forcefully caused to fail). But I'm ok to fix it later in another issue.

Reply:
I created https://github.com/eclipse-thingweb/node-wot/issues/1299